opt: provided ordering not trimmed according to FDs

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3935460&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3935460&tab=artifacts#/) on release-21.2 @ [999d35029f33a9b35dde0599c952efc83ed726a7](https://github.com/cockroachdb/cockroach/commits/999d35029f33a9b35dde0599c952efc83ed726a7):

Random syntax error:

```
    rsg_test.go:779: Crash detected: server panic: pq: internal error: provided -3,-2 can be trimmed to -3 (FDs: key(9); (2)-->(1,3), (7,8)-->(9), (9)-->(7,8), (8)-->(9), (1)==(7), (7)==(1), (2)==(8), (8)==(2), (3)==(9), (9)==(3))
```
Query:

```
        SELECT
        	'22:55:42.968787+06:54:00':::TIMETZ AS col_136333,
        	79112842112805689.30:::DECIMAL AS col_136334,
        	NULL AS col_136335,
        	mod(tab_62191.crdb_internal_mvcc_timestamp::DECIMAL, tab_62191.crdb_internal_mvcc_timestamp::DECIMAL)::DECIMAL
        		AS col_136336
        FROM
        	defaultdb.public.table3@[0] AS tab_62190
        	JOIN defaultdb.public.table3@[0] AS tab_62191 ON
        			(tab_62190.col3_0) = (tab_62191.col3_0)
        			AND (tab_62190.crdb_internal_idx_expr) = (tab_62191.crdb_internal_idx_expr)
        			AND (tab_62190.col3_2) = (tab_62191.crdb_internal_idx_expr)
        			AND (tab_62190.col3_1) = (tab_62191.col3_1)
        			AND (tab_62190.col3_2) = (tab_62191.col3_2)
        WHERE
        	EXISTS(
        		SELECT
        			tab_62192._bytes AS col_136330,
        			COALESCE(
        				ARRAY['06:24:43.613761':::TIME,'23:05:25.549401':::TIME,'00:00:00':::TIME,'23:38:39.079484':::TIME],
        				ARRAY['15:03:29.099292':::TIME,'11:15:50.021786':::TIME,'10:15:19.829471':::TIME]
        			)
        				AS col_136331,
        			'0101000020E6100000AC2939F1714F45C0FAB2EC6D233741C0':::GEOGRAPHY AS col_136332
        		FROM
        			defaultdb.public.seed@seed__int8__float8__date_idx AS tab_62192
        		ORDER BY
        			tab_62192._timestamp, tab_62192._enum, tab_62192._bool
        		LIMIT
        			3:::INT8
        	)
        ORDER BY
        	tab_62191.col3_2 DESC, tab_62191.col3_1 DESC
        LIMIT
        	56:::INT8;
```
Schema:

```
    rsg_test.go:596: To reproduce, use schema:
    rsg_test.go:598: 
        SET CLUSTER SETTING sql.defaults.drop_enum_value.enabled = true;
        SET enable_drop_enum_value = true;
        
        		SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;
        		SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;
        		SET CLUSTER SETTING sql.defaults.interleaved_tables.enabled = true;
        	CREATE TABLE table1 (col1_0 INET NOT NULL, col1_1 REGNAMESPACE, col1_2 REGPROCEDURE, col1_3 OID NOT NULL, col1_4 BIT(40) NULL, col1_5 REGPROCEDURE, col1_6 TIMESTAMPTZ NULL, col1_7 STRING AS (lower(CAST(col1_5 AS STRING))) VIRTUAL, UNIQUE (lower(CAST(col1_4 AS STRING)) ASC, col1_0 DESC, col1_1 ASC, lower(CAST(col1_2 AS STRING)) DESC) STORING (col1_3, col1_6) WHERE (table1.col1_6 < '294276-12-31 23:59:59.999999+00:00':::TIMESTAMPTZ) OR (table1.col1_7 = e'\x00':::STRING), UNIQUE (col1_4 ASC, col1_1 DESC, col1_3 DESC, col1_2 DESC), INDEX (col1_6 DESC, col1_3 DESC), INDEX (col1_7 DESC, col1_6 ASC, col1_0, col1_3 ASC, col1_2, col1_5, col1_4 ASC, col1_1 ASC) WHERE table1.col1_6 = '3000-01-01 00:00:00+00:00':::TIMESTAMPTZ, INDEX (col1_2, col1_3, col1_6 ASC, col1_4 ASC, col1_0 ASC, col1_7, col1_5) STORING (col1_1) WHERE (table1.col1_6 < '-2000-01-01 00:00:00+00:00':::TIMESTAMPTZ) OR (table1.col1_7 < '':::STRING), INDEX (col1_1 DESC, col1_3 DESC, col1_2 ASC) WHERE (table1.col1_6 < '-2000-01-01 00:00:00+00:00':::TIMESTAMPTZ) AND (table1.col1_7 < 'X':::STRING), UNIQUE (col1_0 ASC, col1_5 ASC, col1_2, col1_7 DESC, col1_3 ASC, col1_6, col1_1 ASC) WHERE (table1.col1_7 < e'\U00002603':::STRING) OR (table1.col1_6 > '-2000-01-01 00:00:00+00:00':::TIMESTAMPTZ), INDEX (col1_4 DESC, col1_0 ASC, col1_7 DESC, col1_1 ASC, col1_5, col1_2) STORING (col1_3), FAMILY (col1_3, col1_6, col1_0, col1_2, col1_1), FAMILY (col1_4), FAMILY (col1_5));
        CREATE TABLE table2 (col2_0 TIMESTAMPTZ NOT NULL, col2_1 CHAR, PRIMARY KEY (col2_0), INDEX (col2_0 DESC));
        CREATE TABLE table3 (col3_0 NAME NOT NULL, col3_1 TIMETZ NOT NULL, PRIMARY KEY (col3_1 DESC, col3_0 ASC), col3_2 STRING NOT NULL AS (lower(CAST(col3_1 AS STRING))) VIRTUAL, UNIQUE (col3_2 ASC, col3_1 DESC, col3_0) WHERE (table3.col3_2 != e'\U00002603':::STRING) AND (table3.col3_1 < '24:00:00-15:59:00':::TIMETZ), UNIQUE (col3_2 DESC), INDEX (col3_1), INDEX (col3_0 ASC) WHERE ((table3.col3_0 < e'\x00':::STRING) OR (table3.col3_2 > '"':::STRING)) AND (table3.col3_1 = '00:00:00+15:59:00':::TIMETZ), INDEX (col3_2) WHERE ((table3.col3_2 <= e'\x00':::STRING) OR (table3.col3_1 != '24:00:00-15:59:00':::TIMETZ)) OR (table3.col3_0 = e'\x00':::STRING), UNIQUE (lower(col3_0), col3_2) WHERE ((table3.col3_2 <= '"':::STRING) OR (table3.col3_0 < e'\'':::STRING)) OR (table3.col3_1 > '00:00:00+15:59:00':::TIMETZ), UNIQUE (col3_2));
        ALTER TABLE table1 INJECT STATISTICS e'[{"columns": ["col1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 203154099041071784, "histo_buckets": [{"distinct_range": 0, "num_eq": 8839436645607155193, "num_range": 0, "upper_bound": "0"}, {"distinct_range": 9000000, "num_eq": 1449240156004035485, "num_range": 9000000, "upper_bound": "587302789"}, {"distinct_range": 0, "num_eq": 20000000, "num_range": 80, "upper_bound": "648126068"}, {"distinct_range": 6484518104864779000, "num_eq": 7205831504367068753, "num_range": 6484518104864779006, "upper_bound": "1095364210"}, {"distinct_range": 100, "num_eq": 3483719909153921739, "num_range": 100, "upper_bound": "1568230305"}, {"distinct_range": 7254249215106451000, "num_eq": 2223168367572096120, "num_range": 7254249215106451008, "upper_bound": "1806148528"}, {"distinct_range": 6000000, "num_eq": 9000000000, "num_range": 6000000, "upper_bound": "2983203132"}, {"distinct_range": 1000000000, "num_eq": 70000000000, "num_range": 1000000000, "upper_bound": "3916180969"}, {"distinct_range": 4550630896195516400, "num_eq": 1599746123591059754, "num_range": 4550630896195516297, "upper_bound": "4004240950"}], "histo_col_type": "REGPROCEDURE", "name": "__auto__", "null_count": 2970311126792695489, "row_count": 5820028783959643288}, {"columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 197758722639907406, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5820028783959643288}, {"columns": ["col1_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4737093043868157107, "histo_buckets": [{"distinct_range": 0, "num_eq": 70000000000, "num_range": 0, "upper_bound": "0000000000000000000000000000000000000000"}, {"distinct_range": 200, "num_eq": 5444807676850745715, "num_range": 200, "upper_bound": "0000010010111101011110010010000101100010"}, {"distinct_range": 0, "num_eq": 6901600730062061861, "num_range": 70, "upper_bound": "0000100001100001001001111101010010011001"}, {"distinct_range": 0, "num_eq": 3985514519801618553, "num_range": 90000000000, "upper_bound": "1010000000011100011001000101101011011010"}, {"distinct_range": 0, "num_eq": 100000000000, "num_range": 50000, "upper_bound": "1100000001100101101110010001101001111110"}, {"distinct_range": 218900.08342781282, "num_eq": 5569959658021120122, "num_range": 1000000, "upper_bound": "1110100010000010011110010000010100100110"}, {"distinct_range": 3804135101981358600, "num_eq": 200, "num_range": 4647786233869827919, "upper_bound": "1111001011000100101001100010110000110111"}, {"distinct_range": 1544774336389958000, "num_eq": 100, "num_range": 2762372811066043760, "upper_bound": "1111110111110000110001010001110110001101"}], "histo_col_type": "BIT(40)", "name": "__auto__", "null_count": 972242491509980120, "row_count": 5820028783959643288}, {"columns": ["col1_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3842532116562962937, "histo_col_type": "", "name": "__auto__", "null_count": 2678041390517138979, "row_count": 5820028783959643288}, {"columns": ["col1_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5247135882317191098, "histo_col_type": "TIMESTAMPTZ", "name": "__auto__", "null_count": 3396801791441798013, "row_count": 5820028783959643288}, {"columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 419252218545787916, "histo_buckets": [{"distinct_range": 0, "num_eq": 2815320931078774761, "num_range": 0, "upper_bound": ""}, {"distinct_range": 0, "num_eq": 5006869432253594341, "num_range": 1574440330070605334, "upper_bound": "\\u0017~\\u0012``!q;"}, {"distinct_range": 0, "num_eq": 6000000000, "num_range": 3000, "upper_bound": "\\u001e\\u0007\\u0014"}, {"distinct_range": 0, "num_eq": 712394906072423268, "num_range": 0, "upper_bound": "P;bqO>p\\u001d"}], "histo_col_type": "STRING", "name": "__auto__", "null_count": 1439741246449941439, "row_count": 5820028783959643288}, {"columns": ["col1_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4395290000246681934, "histo_buckets": [{"distinct_range": 0, "num_eq": 7741481904766081747, "num_range": 0, "upper_bound": "112.17.236.18/2"}, {"distinct_range": 3210389378252624400, "num_eq": 5481496839090440599, "num_range": 3210389378252624183, "upper_bound": "55.225.177.112/3"}, {"distinct_range": 0, "num_eq": 2733290839103427816, "num_range": 70000000000, "upper_bound": "44.196.177.72/7"}, {"distinct_range": 829458345565929000, "num_eq": 5406606329762847590, "num_range": 829458345565928943, "upper_bound": "79.40.125.196/20"}, {"distinct_range": 0, "num_eq": 3746718347796721520, "num_range": 60000000000, "upper_bound": "90.44.81.111/20"}, {"distinct_range": 5337630591241210000, "num_eq": 0, "num_range": 5337630591241209877, "upper_bound": "7.215.6.109/26"}, {"distinct_range": 535591554784107260, "num_eq": 1019461656908102247, "num_range": 656180480465958781, "upper_bound": "64db:1e83:3568:6c10:4dfa:1902:c9a:220f/45"}], "histo_col_type": "INET", "name": "__auto__", "null_count": 0, "row_count": 5820028783959643288}, {"columns": ["col1_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5499671016204460916, "histo_col_type": "", "name": "__auto__", "null_count": 3403068340285081223, "row_count": 5820028783959643288}]':::JSONB;
        ALTER TABLE table2 INJECT STATISTICS '[{"columns": ["col2_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 87851715, "histo_col_type": "", "name": "__auto__", "null_count": 58445350, "row_count": 90000000}, {"columns": ["col2_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 71505303, "histo_buckets": [{"distinct_range": 0, "num_eq": 2899969143188332192, "num_range": 0, "upper_bound": "1971-05-20 08:36:34.000915+00:00"}, {"distinct_range": 0, "num_eq": 100000, "num_range": 0, "upper_bound": "2013-07-31 17:54:31.000675+00:00"}, {"distinct_range": 0, "num_eq": 708860491330729325, "num_range": 9000000000, "upper_bound": "2028-12-17 15:20:05.000345+00:00"}], "histo_col_type": "TIMESTAMPTZ", "name": "__auto__", "null_count": 0, "row_count": 90000000}]':::JSONB;
        ALTER TABLE table3 INJECT STATISTICS e'[{"columns": ["col3_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2178095377734522832, "histo_buckets": [{"distinct_range": 0, "num_eq": 70000000, "num_range": 0, "upper_bound": "04:26:21.775585+13:42:00"}, {"distinct_range": 0, "num_eq": 0, "num_range": 3854986152353208573, "upper_bound": "05:46:12.828652+02:48:00"}, {"distinct_range": 414365.1827697958, "num_eq": 2600350693243437743, "num_range": 500000, "upper_bound": "06:31:11.459197-08:02:00"}, {"distinct_range": 0, "num_eq": 1000000000, "num_range": 0, "upper_bound": "10:40:28.043045-04:24:00"}, {"distinct_range": 10000000, "num_eq": 9000000, "num_range": 10000000, "upper_bound": "11:45:13.657638-05:24:00"}, {"distinct_range": 561725838158044100, "num_eq": 600, "num_range": 561725838158044082, "upper_bound": "24:00:00-15:59:00"}], "histo_col_type": "TIMETZ", "name": "__auto__", "null_count": 0, "row_count": 4935508558953418311}, {"columns": ["col3_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3970487219465910087, "histo_buckets": [{"distinct_range": 0, "num_eq": 3000, "num_range": 0, "upper_bound": "\\u0019\\u0012\\u000c"}, {"distinct_range": 0, "num_eq": 0, "num_range": 600000000, "upper_bound": "T\\u0012"}, {"distinct_range": 548290012.4926457, "num_eq": 6111439017269784487, "num_range": 1000000000, "upper_bound": "W"}, {"distinct_range": 0, "num_eq": 2607818290397118812, "num_range": 20, "upper_bound": "X"}, {"distinct_range": 0, "num_eq": 7054515482558932702, "num_range": 2013703180030526118, "upper_bound": "gs~dJ_\\u001f3"}], "histo_col_type": "STRING", "name": "__auto__", "null_count": 0, "row_count": 4935508558953418311}, {"columns": ["col3_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2725371525118870692, "histo_buckets": [{"distinct_range": 0, "num_eq": 3728569627368161633, "num_range": 0, "upper_bound": ""}, {"distinct_range": 50, "num_eq": 100000, "num_range": 50, "upper_bound": "5Z\\u000c+E"}, {"distinct_range": 0, "num_eq": 450689184747613903, "num_range": 0, "upper_bound": "@pz"}, {"distinct_range": 667894147979973100, "num_eq": 60000000000, "num_range": 8625647586004077668, "upper_bound": "X"}, {"distinct_range": 1740070526825056800, "num_eq": 600000000, "num_range": 3358698148713310429, "upper_bound": "c\\u0011O\\ry\\\\"}, {"distinct_range": 4542010005.905104, "num_eq": 8971024423084782046, "num_range": 5000000000, "upper_bound": "v8hh"}], "histo_col_type": "NAME", "name": "__auto__", "null_count": 0, "row_count": 4935508558953418311}]':::JSONB;
        CREATE TYPE rand_typ_0 AS ENUM ('fd');
        CREATE TYPE rand_typ_1 AS ENUM ('kdp', 'ibshde', 'zqp', 'kpqobn', 'mvsk', 'rxqsi');
        CREATE TYPE rand_typ_2 AS ENUM ('xdmr', 'wryu', 'knfq');
        ;
    rsg_test.go:598: 
        SET CLUSTER SETTING sql.defaults.drop_enum_value.enabled = true;
        SET enable_drop_enum_value = true;
        
        CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning');
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:600: 
    rsg_test.go:601: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRandomSyntaxSQLSmith018018660
--- FAIL: TestRandomSyntaxSQLSmith (335.35s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestRandomSyntaxSQLSmith PKG=./pkg/sql/tests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #70663 sql/tests: TestRandomSyntaxSQLSmith failed [C-test-failure O-robot T-sql-queries branch-master]
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
mjibson marked as alumn{us/a}; resolving to rafiss instead
```
</p>
</details>

/cc @cockroachdb/sql-experience rafiss
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

_Originally posted by @cockroach-teamcity in https://github.com/cockroachdb/cockroach/issues/70855#issuecomment-996478927_

--- a/pkg/sql/opt/ordering/lookup_join.go
+++ b/pkg/sql/opt/ordering/lookup_join.go
@@ -64,7 +64,16 @@ func lookupOrIndexJoinBuildChildReqOrdering(
 	//
 	// This case indicates that we didn't do a good job pushing down equalities
 	// (see #36219), but it should be handled correctly here nevertheless.
-	return trimColumnGroups(&res, &child.Relational().FuncDeps)
+	res = trimColumnGroups(&res, &child.Relational().FuncDeps)
+
+	// The propagation of FDs might not be perfect; in that case we need to
+	// simplify the required ordering, or risk passing through unnecessary columns
+	// in provided orderings.
+	if fds := &child.Relational().FuncDeps; res.CanSimplify(fds) {
+		res = res.Copy()
+		res.Simplify(fds)
+	}
+	return res
 }
 
 func indexJoinBuildProvided(expr memo.RelExpr, required *props.OrderingChoice) opt.Ordering {

--- a/pkg/sql/opt/xform/testdata/external/trading
+++ b/pkg/sql/opt/xform/testdata/external/trading
@@ -714,7 +714,7 @@ project
       │    │    ├── stats: [rows=478.646617]
       │    │    ├── key: (3,5)
       │    │    ├── fd: ()-->(1,2,4,20,21), (3,5)-->(6,7)
-      │    │    ├── ordering: -3 opt(1,2,4) [actual: -3]
+      │    │    ├── ordering: -3 opt(1,2,4,20,21) [actual: -3]
       │    │    ├── limit hint: 100.00
       │    │    ├── index-join transactiondetails
       │    │    │    ├── columns: transactiondetails.dealerid:1!null transactiondetails.isbuy:2!null transactiondate:3!null cardid:4!null quantity:5!null sellprice:6!null buyprice:7!null
@@ -848,7 +848,7 @@ project
  │         │    │    │    ├── stats: [rows=19000]
  │         │    │    │    ├── key: (1)
  │         │    │    │    ├── fd: ()-->(42,43), (1)-->(2-6), (2,4,5)~~>(1,3,6)
- │         │    │    │    ├── ordering: +1
+ │         │    │    │    ├── ordering: +1 opt(42,43) [actual: +1]
  │         │    │    │    ├── select
  │         │    │    │    │    ├── columns: id:1!null name:2!null rarity:3 setname:4 number:5!null isfoil:6!null
  │         │    │    │    │    ├── immutable

--- a/pkg/sql/opt/xform/testdata/external/trading-mutation
+++ b/pkg/sql/opt/xform/testdata/external/trading-mutation
@@ -718,7 +718,7 @@ project
       │    │    ├── stats: [rows=478.646617]
       │    │    ├── key: (3,5)
       │    │    ├── fd: ()-->(1,2,4,24,25), (3,5)-->(6,7)
-      │    │    ├── ordering: -3 opt(1,2,4) [actual: -3]
+      │    │    ├── ordering: -3 opt(1,2,4,24,25) [actual: -3]
       │    │    ├── limit hint: 100.00
       │    │    ├── index-join transactiondetails
       │    │    │    ├── columns: transactiondetails.dealerid:1!null transactiondetails.isbuy:2!null transactiondate:3!null cardid:4!null quantity:5!null sellprice:6!null buyprice:7!null
@@ -852,7 +852,7 @@ project
  │         │    │    │    ├── stats: [rows=19000]
  │         │    │    │    ├── key: (1)
  │         │    │    │    ├── fd: ()-->(48,49), (1)-->(2-6), (2,4,5)~~>(1,3,6)
- │         │    │    │    ├── ordering: +1
+ │         │    │    │    ├── ordering: +1 opt(48,49) [actual: +1]
  │         │    │    │    ├── select
  │         │    │    │    │    ├── columns: id:1!null name:2!null rarity:3 setname:4 number:5!null isfoil:6!null
  │         │    │    │    │    ├── immutable

--- a/pkg/sql/opt/xform/testdata/physprops/ordering
+++ b/pkg/sql/opt/xform/testdata/physprops/ordering
@@ -2398,7 +2398,7 @@ distinct-on
       ├── project
       │    ├── columns: "lookup_join_const_col_@9":14!null a:1!null b:5!null
       │    ├── fd: ()-->(14), (1)==(5), (5)==(1)
-      │    ├── ordering: +(1|5) [actual: +1]
+      │    ├── ordering: +(1|5) opt(14) [actual: +1]
       │    ├── inner-join (lookup t44469_b@t44469_b_b_idx)
       │    │    ├── columns: a:1!null b:5!null
       │    │    ├── flags: force lookup join (into right side)
@@ -2448,3 +2448,113 @@ project
  │              └── k:1 > 0 [outer=(1)]
  └── projections
       └── 1 [as="?column?":6]
+
+# Regression test for #73968. Lookup join needs to simplify the ordering
+# required of its child.
+exec-ddl
+CREATE TABLE t73968 (
+  k INT PRIMARY KEY,
+  name STRING,
+  x STRING AS (CAST(k AS STRING)) VIRTUAL,
+  y STRING AS (lower(name)) VIRTUAL,
+  UNIQUE (y),
+  UNIQUE (x)
+);
+----
+
+exec-ddl
+ALTER TABLE t73968 INJECT STATISTICS e'[
+  {
+    "columns": ["k"],
+    "created_at": "2000-01-01 00:00:00+00:00",
+    "distinct_count": 1000000,
+    "name": "__auto__",
+    "null_count": 0,
+    "row_count": 1000000000
+  }
+]';
+----
+
+opt
+SELECT
+  t2.crdb_internal_mvcc_timestamp
+FROM
+  t73968 AS t1 JOIN t73968 AS t2 ON t1.name = t2.name
+      AND t1.y = t2.y
+      AND t1.x = t2.y
+      AND t1.k = t2.k
+      AND t1.x = t2.x
+ORDER BY
+  t2.x, t2.k
+LIMIT
+  56
+----
+project
+ ├── columns: crdb_internal_mvcc_timestamp:11  [hidden: k:7!null x:9!null]
+ ├── cardinality: [0 - 56]
+ ├── immutable
+ ├── key: (7)
+ ├── fd: (7)-->(9,11)
+ ├── ordering: +9,+7 [actual: +9]
+ └── limit
+      ├── columns: k:1!null name:2!null x:3!null k:7!null name:8!null x:9!null crdb_internal_mvcc_timestamp:11
+      ├── internal-ordering: +(3|9),+(1|7)
+      ├── cardinality: [0 - 56]
+      ├── immutable
+      ├── key: (7)
+      ├── fd: (1)-->(2,3), (7)-->(8,9,11), (2)==(8), (8)==(2), (1)==(7), (7)==(1), (3)==(9), (9)==(3)
+      ├── ordering: +(3|9),+(1|7) [actual: +3]
+      ├── inner-join (lookup t73968)
+      │    ├── columns: k:1!null name:2!null x:3!null k:7!null name:8!null x:9!null crdb_internal_mvcc_timestamp:11
+      │    ├── key columns: [7] = [7]
+      │    ├── lookup columns are key
+      │    ├── immutable
+      │    ├── key: (7)
+      │    ├── fd: (1)-->(2,3), (7)-->(8,9,11), (2)==(8), (8)==(2), (1)==(7), (7)==(1), (3)==(9), (9)==(3)
+      │    ├── ordering: +(3|9),+(1|7) [actual: +3]
+      │    ├── limit hint: 56.00
+      │    ├── inner-join (lookup t73968@t73968_x_key)
+      │    │    ├── columns: k:1!null name:2 x:3!null k:7!null x:9!null
+      │    │    ├── key columns: [3] = [9]
+      │    │    ├── lookup columns are key
+      │    │    ├── immutable
+      │    │    ├── key: (7)
+      │    │    ├── fd: (1)-->(2,3), (7)-->(9), (9)-->(7), (1)==(7), (7)==(1), (3)==(9), (9)==(3)
+      │    │    ├── ordering: +(3|9) [actual: +3]
+      │    │    ├── limit hint: 200.00
+      │    │    ├── sort
+      │    │    │    ├── columns: k:1!null name:2 x:3!null
+      │    │    │    ├── immutable
+      │    │    │    ├── key: (1)
+      │    │    │    ├── fd: (1)-->(2,3)
+      │    │    │    ├── ordering: +3
+      │    │    │    ├── limit hint: 2100.00
+      │    │    │    └── project
+      │    │    │         ├── columns: x:3!null k:1!null name:2
+      │    │    │         ├── immutable
+      │    │    │         ├── key: (1)
+      │    │    │         ├── fd: (1)-->(2,3)
+      │    │    │         ├── select
+      │    │    │         │    ├── columns: k:1!null name:2
+      │    │    │         │    ├── immutable
+      │    │    │         │    ├── key: (1)
+      │    │    │         │    ├── fd: (1)-->(2)
+      │    │    │         │    ├── scan t73968
+      │    │    │         │    │    ├── columns: k:1!null name:2
+      │    │    │         │    │    ├── computed column expressions
+      │    │    │         │    │    │    ├── x:3
+      │    │    │         │    │    │    │    └── k:1::STRING
+      │    │    │         │    │    │    └── y:4
+      │    │    │         │    │    │         └── lower(name:2)
+      │    │    │         │    │    ├── key: (1)
+      │    │    │         │    │    └── fd: (1)-->(2)
+      │    │    │         │    └── filters
+      │    │    │         │         └── k:1::STRING = lower(name:2) [outer=(1,2), immutable]
+      │    │    │         └── projections
+      │    │    │              └── k:1::STRING [as=x:3, outer=(1), immutable]
+      │    │    └── filters
+      │    │         └── k:1 = k:7 [outer=(1,7), fd=(1)==(7), (7)==(1)]
+      │    └── filters
+      │         ├── name:2 = name:8 [outer=(2,8), fd=(2)==(8), (8)==(2)]
+      │         └── k:7::STRING = lower(name:8) [outer=(7,8), immutable]
+      └── 56

--- a/pkg/sql/opt/xform/testdata/rules/groupby
+++ b/pkg/sql/opt/xform/testdata/rules/groupby
@@ -2212,14 +2212,14 @@ memo (optimized, ~32KB, required=[presentation: w:12] [ordering: +13,+1])
  ├── G10: (filters G19)
  ├── G11: (filters)
  ├── G12: (project G8 G20 x)
- │    ├── [ordering: +1]
+ │    ├── [ordering: +1 opt(14)]
  │    │    ├── best: (project G8="[ordering: +1]" G20 x)
  │    │    └── cost: 1074.34
  │    └── []
  │         ├── best: (project G8 G20 x)
  │         └── cost: 1074.34
  ├── G13: (project G8 G20 x)
- │    ├── [ordering: +1]
+ │    ├── [ordering: +1 opt(15)]
  │    │    ├── best: (project G8="[ordering: +1]" G20 x)
  │    │    └── cost: 1074.34
  │    └── []

